runtime/doc/filetype.txt

```diff
@@ -176,125 +176,16 @@ This means that the contents of compressed files are not inspected.
 							*new-filetype*
 If a file type that you want to use is not detected yet, there are a few ways
 to add it.  In any way, it's better not to modify the $VIMRUNTIME/filetype.lua
-or $VIMRUNTIME/filetype.vim files.  They will be overwritten when installing a
-new version of Nvim. The following explains the legacy Vim mechanism (enabled
-if |g:do_legacy_filetype| is set). For Nvim's default mechanism, see
-|vim.filetype.add()|.
-
-A. If you want to overrule all default file type checks.
-   This works by writing one file for each filetype.  The disadvantage is that
-   there can be many files.  The advantage is that you can simply drop this 
-   file in the right directory to make it work.
-							*ftdetect*
-   1. Create your user runtime directory.  You would normally use the first
-      item of the 'runtimepath' option.  Then create the directory "ftdetect"
-      inside it.  Example for Unix: >
-	:!mkdir -p ~/.config/nvim/ftdetect
-<
-   2. Create a file that contains an autocommand to detect the file type.
-      Example: >
-	au BufRead,BufNewFile *.mine		set filetype=mine
-<     Note that there is no "augroup" command, this has already been done
-      when sourcing your file.  You could also use the pattern "*" and then
-      check the contents of the file to recognize it.
-      Write this file as "mine.vim" in the "ftdetect" directory in your user
-      runtime directory.  For example, for Unix: >
-	:w ~/.config/nvim/ftdetect/mine.vim
```

Comment:
I don't see this opt-in runtimepath approach as deprecated. Just like Lua plugins, we might also look for `ftdetect/foo.lua`, but the general idea of passively discovering overrides on the user's config path seems like a standard approach that should remain.

Though certainly this deleted area can be greatly abbreviated.

Reply:
> Though certainly this deleted area can be greatly abbreviated.

It can, but at the price that we now "own" it. I'd prefer to not spend much time rewriting this legacy section and instead leave it as-is.